### ISSUE TITLE
mysql_user doesn't accept encryption

### ISSUE THREAD

**Issue by cortopy** [ansible/ansible-modules-core#1662]
I normally use Passlib encryption with the prompt module in order to set passwords. This works well with SSH passwords, for example.

However, when trying to use prompt to set the password variable with mysql_user it just doesn't work. I've tried with sha512_crypt and md5_crypt but mysql wouldn't let me log in after changing passwords. 

Using prompt without encryption works fine. 

Is this because mysql_user is not compatible with passlib?

Just mentioning @marktheunissen as it appears as author of the module

**Comment by Jmainguy** [comment]
@cortopy pretend I am an newbie, could you show an example of your desired work flow for this.

Are you passing an encrypted string to mysql_user for the password and is rejecting it?

**Comment by cortopy** [comment]
That's exactly it. 

I would the encryption workflow with prompt as described in http://docs.ansible.com/ansible/playbooks_prompts.html. My guess is that mysql is encrypting the already encrypted password. 

Since this wasn't working for me I changed my workflow to just using Vault. I've not had any problems with it so far.

**Comment by Jmainguy** [comment]
Yeah I think Vault is likely the best solution for this. You open to closing this, or would you still like to see this get implemented?

**Comment by cortopy** [comment]
Perhaps it may be a good idea to replicate what the user module docs say for password parameter: "See http://docs.ansible.com/ansible/faq.html#how-do-i-generate-crypted-passwords-for-the-user-module for details on various ways to generate these password values. Note on Darwin system, this value has to be cleartext. Beware of security issues."

I don't know if this would be possible but couldn't find the docs in github to request pull

**Comment by ansibot** [comment]
@Jmainguy, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by Jmainguy** [comment]
@cortopy Sorry for the delayed response.

mysql_user now allows encrypted as an option, if you know the encrpyed hash mysql would use for the password, you can now supply that as an option and it wont encrypt it a second time. Can you test that and see if it meets your needs, or if we still need to add another feature here?

**Comment by ansibot** [comment]
@Jmainguy, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Jmainguy** [comment]
ansibot needs_info

**Comment by cortopy** [comment]
thanks @Jmainguy !
Unfortunately I don't have access to a mysql server that's up and running to quickly test this. What you say sounds about right

**Comment by thorian93** [comment]
I am unsure whether I missed something, but I could not find any information in the documentation on how to create a hash for mysql like there is for the user module. I will look it up now, but for the future a hint in the documentation as @cortopy suggested in [this comment](https://github.com/ansible/ansible-modules-core/issues/1662#issuecomment-162631582) would be great.

**edit:** I looked up how to create a prehashed password and as far as I can tell, the only way to do is to log into an existing mysql instance and create the hash using the built-in `SELECT SHA1(UNHEX(SHA1("password")));` algorithm. Is that correct or are there other ways?

I know this is out of scope for the issue but maybe we can prevent other users from doing hour-long research on how to achieve what they want by correcting this now.

**Comment by ansibot** [comment]
@Jmainguy, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/30093
